rustfmt-core/rustfmt-lib/src/visitor.rs

```diff
@@ -594,67 +597,95 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
-            ast::ImplItemKind::TyAlias(ref ty) => {
-                let rewrite = rewrite_associated_impl_type(
-                    ii.ident,
-                    ii.defaultness,
-                    Some(ty),
-                    &ii.generics,
-                    &self.get_context(),
-                    self.block_indent,
-                );
+            ast::AssocItemKind::Fn(ref sig, None) => {
+                let indent = self.block_indent;
+                let rewrite =
+                    self.rewrite_required_fn(indent, ii.ident, sig, &ii.generics, ii.span);
                 self.push_rewrite(ii.span, rewrite);
             }
-            ast::ImplItemKind::OpaqueTy(ref generic_bounds) => {
-                let rewrite = rewrite_opaque_impl_type(
-                    &self.get_context(),
-                    ii.ident,
-                    &ii.generics,
-                    generic_bounds,
-                    self.block_indent,
-                );
+            ast::AssocItemKind::Const(..) => self.visit_static(&StaticParts::from_impl_item(ii)),
+            ast::AssocItemKind::TyAlias(_, ref ty) => {
+                let rewrite_associated = || {
+                    rewrite_associated_impl_type(
+                        ii.ident,
+                        ii.defaultness,
+                        ty.as_ref(),
+                        &ii.generics,
+                        &self.get_context(),
+                        self.block_indent,
+                    )
+                };
+                let rewrite = match ty {
+                    None => rewrite_associated(),
+                    Some(ty) => match ty.kind.opaque_top_hack() {
+                        Some(generic_bounds) => rewrite_opaque_impl_type(
+                            &self.get_context(),
+                            ii.ident,
+                            &ii.generics,
+                            generic_bounds,
+                            self.block_indent,
+                        ),
+                        None => rewrite_associated(),
+                    }
+                };
                 self.push_rewrite(ii.span, rewrite);
             }
-            ast::ImplItemKind::Macro(ref mac) => {
+            ast::AssocItemKind::Macro(ref mac) => {
                 self.visit_mac(mac, Some(ii.ident), MacroPosition::Item);
             }
         }
     }
 
     fn visit_mac(&mut self, mac: &ast::Mac, ident: Option<ast::Ident>, pos: MacroPosition) {
-        skip_out_of_file_lines_range_visitor!(self, mac.span);
+        skip_out_of_file_lines_range_visitor!(self, mac.span());
 
         // 1 = ;
         let shape = self.shape().saturating_sub_width(1);
         let rewrite = self.with_context(|ctx| rewrite_macro(mac, ident, ctx, shape, pos));
-        self.push_rewrite(mac.span, rewrite);
+        let (span, rewrite) = match macro_style(mac, &self.get_context()) {
```

Comment:
This is to address the change in this version of the rustc-ap creates where the span no longer includes the associated `;` (https://github.com/rust-lang/rustfmt/pull/4022#issuecomment-574961034). Wasn't sure what the best solution to this would be, so figured I'd try to just get the "correct" span to include the `;` when warranted, so that Mac items that have whitespace between the closing delim and the semicolon are still formatted correctly (i.e. `)       ;` is still formatted to `);`)